### DISCUSSION TITLE
Hide attributes from dataclass?

### DISCUSSION THREAD

**Discussion by Gabriel-p** [sphinx-extensions2/sphinx-autodoc2#63] [Q&A]
I have  classes defined with the `@dataclass` decorator. This means that they look like this:

```
@dataclass
class myclass:
    """Some class

    :param first_param: some description
    :param second_param: some description

    first_param: str
    second_param: int

    def __post_init__(self):
        etc...
```

This makes `autodoc2` document the `first_param` and `second_param` twice within the class: once inside the `Parameters` list, and once below this as individual attributes.

How can I hide the duplicate listing of these parameters?

**Comment by wtwang1998** [comment]
I have the same question. I try to delete docstring and use doc comment like [Doc comments and docstrings](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#doc-comments-and-docstrings), but unfortunately, autodoc2 cannot recognize it.

**Comment by ZaberKo** [comment]
I find a solution, see https://github.com/sphinx-extensions2/sphinx-autodoc2/issues/33#issuecomment-2684177928
